crates/lovely-core/src/patch/regex.rs

```diff
@@ -94,21 +110,82 @@ impl RegexPatch {
                 &mut line_prepend
             );
 
+            // Cleanup and convert the specified root capture to a span.
+            let target_group = {
+                let group_name = self
+                    .root_capture
+                    .as_deref()
+                    .unwrap_or("0")
+                    .replace('$', "");
+
+                if let Ok(idx) = group_name.parse::<usize>() {
+                    groups.get_group(idx)
+                        .unwrap_or_else(|| 
+                            panic!("The capture group at index {idx} could not be found in '{base_str}' with the Regex '{}'", self.pattern))
+                } else {
+                    groups.get_group_by_name(&group_name)
+                        .unwrap_or_else(|| 
+                            panic!("The capture group with name '{group_name}' could not be found in '{base_str}' with the Regex '{}'", self.pattern))
+                }
+            };
+
+            let target_start = (target_group.start as isize + delta) as usize;
+            let target_end = (target_group.end as isize + delta) as usize;
+
+            let mut new_payload = String::new();
+
+            // If left border of insertion is a wordchar -> non-wordchar 
+            // boundary and our patch starts with a wordchar, prepend space so 
+            // it doesn't unintentionally concatenate with characters to its 
+            // left to create a larger identifier.
+            if self.payload.starts_with(|x: char| x.is_ascii_alphanumeric() || x == '_' || x == '$' || x == '{' || x == '}') {
+                let pre_pt = if let InsertPosition::After = self.position {
+                    target_end
+                } else {
+                    target_start
+                };
+                if pre_pt > 0 {
+                    let byte_on_left = rope.byte(pre_pt - 1);
+                    if byte_on_left.is_ascii_alphanumeric() || byte_on_left == b'_' {
+                        new_payload.push(' ');
+                    }
+                }
+            }
+
             // Prepend each line of the payload with line_prepend.
-            let new_payload = self
-                .payload
-                .lines()
-                .map(|x| if !x.is_empty() {
-                    format!("{line_prepend}{x}")
+            std::fmt::write(
```

Comment:
The example in rust docs uses format!, that's simpler

Reply:
Okay, changed.